discussion_notes.md

```diff
@@ -0,0 +1,104 @@
+- Longer context lengths can amplify the impact of concept drift. As the model tries to capture a broader range of historical patterns, it might struggle to adapt to the changing dynamics, especially when the prediction length remains constant.
+- In some cases, longer context lengths can lead to overfitting, especially if the model is complex. Data Imbalance: With a higher context-to-prediction ratio, the model is exposed to an imbalanced dataset. The model might overfit to the context data, leading to poor generalization and reduced performance on the prediction task.
+- Lack of Generalization: The model might become overly specific to the context data, failing to generalize to the prediction task. Put another way, the increased context data might be inconsistent with the actuals in the prediction window. 
+- Evaluation Metric Sensitivity: The choice of evaluation metrics can also influence the perceived performance. Some metrics might be more sensitive to changes in the context-to-prediction ratio, leading to a perceived decrease in performance
+	- WQL's sensitivity to the context-to-prediction ratio can arise when the distribution of errors changes with varying ratios. For example, if the context length increases, the distribution of errors might shift, leading to different quantile values and, consequently, different WQL values. This can make it challenging to directly compare WQL values across different ratios, especially if the underlying data patterns are complex or non-stationary. [citation needed]
+
+
+**What do good scores on our error metrics mean? What does it mean for WQL, EMD or MASE to be low, respectively?**
+
+Low mWQL suggests that the model can capture the distribution of errors effectively, regardless of the quantile level.
+
+A good score on mWQL means:
+- The model's forecasts are consistently accurate across different quantile levels. 
+- The model can handle a variety of error distributions and is robust to changes in the data's underlying patterns.
+- The model's performance is well-calibrated, ensuring that the forecasts are not overly optimistic or pessimistic.
+
+A good score on MASE means:
+- The model's forecasts are accurate and reliable.
+- The model can capture the underlying patterns and trends in the time series data effectively.
+- The model's performance is consistent and robust, even when the data exhibits non-stationary behavior or concept drift.
+
+MASE is particularly useful when comparing the performance of different forecasting models, as it provides a standardized measure of error that is independent of the scale of the data.
+
+
+A low EMD value indicates that the model's forecasts are not only accurate at the mean or median but also capture the spread and shape of the actual values accurately. This metric is valuable for assessing the calibration and overall performance of probabilistic forecasting models.
+
+A low EMD value means:
```

Comment:
OJO: We changed the meaning of "EMD" quite a bit - these changes make EMD very similar to both EMD and MASE -- call it "Mean quantiile scaled EMD"
- we're doing EMD on the mean quantile, which is very similar to mWQL
- When we scale the EMD we're doing the same process MASE does to create scaled error by normalizing based on the seasonal offset

Reply:
- MASE marches along and compares each point in time between the actuals and the forecasts
- EMD marches along and calculates the minimum error between each point and all other points, penalized by how far it must go to find the minimum. 

In theory, EMD should be bounded by the Mean Absolute Error (MAE)